### PR TITLE
[ fix ] : duplicate boards after boardname change  

### LINKED ISSUE #63: Change Board Name not working properly
After changing the board name from Untitled Whiteboard, another board is being added to the persistance database rather than updating the previously rendered board.

https://github.com/RocketChat/Apps.Whiteboard/assets/107868772/173f8634-67aa-42c3-a3b3-35b534f3f4da

### PR DESCRIPTION
**Pull request by brf153**
closes #63 

https://github.com/RocketChat/Apps.Whiteboard/assets/107868772/20a0f80f-522f-41ee-bf1f-21d8dfa5f054

### CODE REVIEW

**Comment by brf153** [comment]
@CulturalProfessor this was the main issue. I have fixed it. First we need to merge this pr and then we can work on the search board command.

**Comment by brf153** [comment]
This pr adds two functionalities:
i) We need to update the boardname and not add another board for it. I have fixed it.

**Initially I was getting boardData like this**
Screenshot from 2023-12-31 13-51-34
 See that we are getting two objects with the same messageId and id which means a new board has been created rather than replacing the previous one.
 
 
**Now, we will get boardData like this**
Screenshot from 2023-12-31 13-50-20


ii) I have also added the functionality that `whiteboard delete _all` will delete all the boards in a room.

**Comment by brf153** [comment]
Also note that I have changed now how the `/whiteboard list` command will be functioning. I have made another association with both the roomname and boardid. This will be unique for a board in the given room. Initially using the list command will render a message stating `Error here messages.ts`. It is because I have made some changes in the association. Make new boards in the room and you would not face the error.

**Comment by CulturalProfessor** [comment]
@brf153 just check in db also if duplicates are not made

**Comment by brf153** [comment]
@CulturalProfessor I wanted to ask that is the data being stored in the persistance the same as the data being stored in the mongodb database? I think it is. I actually sent a screenshot of console logged value. That value is the data being stored in the persistance storage. Also can you please tell me how will I check the mongodb database? I have been quering the persistance database only till now.

**Comment by CulturalProfessor** [comment]
yep they are same, @brf153 install mongodb compass and add uri as `mongodb://localhost:3001/` if server is running on port 3000 and if any port then just add 1 in last digit , in that see apps persistence you'll see the records.

**Comment by brf153** [comment]
@CulturalProfessor after changing the name of the boards, we were not updating the boards but rather making a new one. It is confirmed that we were having multiple boards in the database as well. I think we can proceed with the review of this pr.

**Comment by CulturalProfessor** [comment]
> @CulturalProfessor after changing the name of the boards, we were not updating the boards but rather making a new one. It is confirmed that we were having multiple boards in the database as well. I think we can proceed with the review of this pr.

so now we're not making multiple boards right ?

**Comment by brf153** [comment]
Nope

**Comment by CulturalProfessor** [comment]
> Nope

is your db loading now , share some ss and remove **1** and **2** board code

**Comment by brf153** [comment]
@CulturalProfessor I am still not being able to check my database. I am querying the data of persistence using console log. I told Sumit to check the database. I got this response

WhatsApp Image 2024-01-05 at 15 34 03

`remove 1 and 2 board code` I did not get it. What do I need to remove?

File: whiteboard/handlers/ExecuteViewSubmitHandler.ts [hunk 1]
```diff
@@ -109,6 +109,13 @@ export class ExecuteViewSubmitHandler {
                                 if (checkBoardName == 1) {
                                     checkName = 1;
                                 }
+
```
Comment (CulturalProfessor) [inline whiteboard/handlers/ExecuteViewSubmitHandler.ts:112]: remove this

**Comment by CulturalProfessor** [comment]
@brf153 what's the problem in connecting to db ? is not working by running rc local server

**Comment by CulturalProfessor** [comment]
@brf153 check this out for docker https://docs.rocket.chat/deploy/deploy-rocket.chat/deploy-with-docker-and-docker-compose#docker-mongo-backup-and-restore

**Comment by brf153** [comment]
@CulturalProfessor I am running server using docker compose file(just like they have shown in the youtube video in which they were making the hello world app). Now, when I am trying to connect to the mongodb using mongodb uri `mongodb://localhost:3001/` , it gives me an error.
Screenshot from 2024-01-05 17-21-33

I am running the server on port 3000, still the database is not connecting. I don't know why. I have checked their docker compose file and I have used the uri `mongodb://bhandari:password@mongodb:27017/?authSource=admin&replicaSet=rs0` still, it does not work.

**Comment by CulturalProfessor** [comment]
> @CulturalProfessor oki I will check the rocket chat docker compose docs. WHat should we do about this pr? I think Sumit has confirmed the error. ALso, I have checked the logs in my terminal and it is working fine now after making the changes.

let's wait till your connection works

**Comment by brf153** [comment]
@CulturalProfessor oki. I will try to fix it today.

**Comment by brf153** [comment]
@CulturalProfessor by the way, how are you running the server? Have you cloned the [rocket chat repo](https://github.com/RocketChat/Rocket.Chat) into your system and started the server?

**Comment by CulturalProfessor** [comment]
> @CulturalProfessor by the way, how are you running the server? Have you cloned the [rocket chat repo](https://github.com/RocketChat/Rocket.Chat) into your system and started the server?

Yeah

**Comment by brf153** [comment]
@CulturalProfessor now, I am being able to connect to the database. But how will I get the boards data from it?
Screenshot from 2024-01-05 20-13-41

**Comment by brf153** [comment]
@CulturalProfessor now I am being able to connect to the database. But how will I check the board data in a room?
Screenshot from 2024-01-05 20-34-22

**Comment by CulturalProfessor** [comment]
> @CulturalProfessor now I am being able to connect to the database. But how will I check the board data in a room? Screenshot from 2024-01-05 20-34-22

image
check here

**Comment by brf153** [comment]
@CulturalProfessor see we are getting two boards with same id. I changed the board name from `bhandari` to `toto`
Screenshot from 2024-01-05 21-13-59

**Comment by brf153** [comment]
@CulturalProfessor I have checked the database. This pr fixes the problem. We will modify the boardname rather than creating a new board.
https://github.com/RocketChat/Apps.Whiteboard/assets/107868772/e9cd931b-aaab-4ffe-a255-9c53de165316

**Comment by brf153** [comment]
Is the video visible?

**Comment by brf153** [comment]
https://github.com/RocketChat/Apps.Whiteboard/assets/107868772/55ed2280-e56b-4dde-8ce2-dde0287566e7

**Comment by brf153** [comment]
wait work in progress

**Comment by brf153** [comment]
@CulturalProfessor I have checked the pr. It works perfectly. It works in the whiteboard as well as in the database. ALso, I have kept the `checkName ==2` because we do not want anyone to keep name as `_all` because we are using the command `/whiteboard delete _all` to delete all the boards in the given room. Also, Sumit made the function initially and we are using the `checkName==1` to check whether the name is being repeated or not.

**Comment by CulturalProfessor** [comment]
> @CulturalProfessor I have checked the pr. It works perfectly. It works in the whiteboard as well as in the database. ALso, I have kept the `checkName ==2` because we do not want anyone to keep name as `_all` because we are using the command `/whiteboard delete _all` to delete all the boards in the given room. Also, Sumit made the function initially and we are using the `checkName==1` to check whether the name is being repeated or not.

there'a not any **/whiteboard delete _all** command and please modify checkname to use boolean not 1 or 2

**Comment by brf153** [comment]
actually I made a lot of boards in the room and I wanted to clear all of them. Sometimes it might happen that someone makes a lot of boards by mistake or sometimes some scammers might enter. It might come in handy.

**Comment by brf153** [comment]
@CulturalProfessor I have removed the delete _all command and replaced numbers with boolean for conditional rendering

**Comment by CulturalProfessor** [comment]
@brf153 this pr took a long time but good work

**Comment by brf153** [comment]
@CulturalProfessor yeah but thank you for reviewing the pr. Hope the prior mistakes do not repeat.

**Pull request merged**